shm/src/Game.cpp

```diff
@@ -0,0 +1,409 @@
+#include "shm/inc/Game.hpp"
+
+#include <algorithm>
+#include <cmath>
+#include <cstddef>
+#include <iomanip>
+#include <iostream>
+#include <memory>
+
+#include "shm/inc/Island.hpp"
+
+namespace {
+    constexpr size_t DISTANCE_MULTIPLIER{ 10 };
+    constexpr size_t FIRST_OPTION_ELEMENT{ 1 };
+    constexpr size_t LAST_OPTION_ELEMENT{ 7 };
+}
+
+Game::Game(size_t money, size_t gameDays, size_t finalGoal)
+    : money_(money)
+    , gameDays_(gameDays)
+    , finalGoal_(finalGoal)  
+    , time_(std::make_shared<Time>())
+    , map_(std::make_shared<Map>(time_.get()))
+    , player_(std::make_unique<Player>(std::make_unique<Ship>(1, 25, 100, time_.get(), nullptr),
+                                       map_, 100, 10000, time_.get()))
```

Comment:
make constexpr values for this 1, 25, 100, 10000 etc... It will be easier to replace to other values later